timezone/test/Timezone_test.rb

```diff
@@ -225,4 +228,317 @@
       end
     end
   end
+
+  describe "#GetDateTime" do
+    context "real_time is set to true" do
+      context "locale_format is set to false" do
+        it "returns stripped time output without locale format" do
+          expect(Yast::SCR).to receive(:Execute).with(
+            path(".target.bash_output"),
+            "/bin/date \"+%Y-%m-%d - %H:%M:%S\""
+          ).and_return("stdout" => "2020-02-05 - 08:13:57\n")
+
+          expect(subject.GetDateTime(true, false)).to eq "2020-02-05 - 08:13:57"
+        end
+      end
+
+      context "locale_format is set to true" do
+        it "returns stripped time output with locale format" do
+          expect(Yast::SCR).to receive(:Execute).with(
+            path(".target.bash_output"),
+            "/bin/date \"+%c\""
+          ).and_return("stdout" => "St??5.????nor??2020,??08:18:27??CET\n")
+
+          expect(subject.GetDateTime(true, true)).to eq "St??5.????nor??2020,??08:18:27??CET"
+        end
+      end
+    end
+
+    context "real_time is set to false" do
+      before do
+        subject.diff = 1
+        subject.timezone = ""
+
+        allow(Yast::SCR).to receive(:Execute).with(
+          path(".target.bash_output"),
+          "/bin/date +%z"
+        ).and_return("stdout" => "+0100\n")
+      end
+
+      context "locale_format is set to false" do
+        it "returns stripped time output without locale format including time zone diff" do
+          expect(Yast::SCR).to receive(:Execute).with(
+            path(".target.bash_output"),
+            "/bin/date \"+%Y-%m-%d - %H:%M:%S\" \"--date=now 3600sec\""
+          ).and_return("stdout" => "2020-02-05 - 09:13:57\n")
+
+          expect(subject.GetDateTime(false, false)).to eq "2020-02-05 - 09:13:57"
+        end
+      end
+
+      context "locale_format is set to true" do
+        it "returns stripped time output with locale format" do
+          expect(Yast::SCR).to receive(:Execute).with(
+            path(".target.bash_output"),
+            "/bin/date \"+%c\" \"--date=now 3600sec\""
+          ).and_return("stdout" => "St??5.????nor??2020,??09:18:27??CET\n")
```

Comment:
More strange ?uestion ma?ks

Reply:
well, it should be UTF8. Probably my locale is broken after testing yast2 CLI for language :)